zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -276,42 +276,41 @@ object JsonCodec {
 
     private def enumEncoder[Z](parentSchema: Schema.Enum[Z], cases: Schema.Case[Z, _]*): ZJsonEncoder[Z] =
       (value: Z, indent: Option[Int], out: Write) => {
-        val nonTransientCases = cases.filter {
-          case Schema.Case(_, _, _, _, _, annotations) =>
-            annotations.collectFirst { case _: transientCase => () }.isEmpty
+        val nonTransientCase = cases.collectFirst {
+          case c @ Schema.Case(_, _, _, _, _, annotations)
+              if annotations.collectFirst { case _: transientCase => () }.isEmpty &&
+                c.deconstructOption(value).isDefined =>
+            c
```

Comment:
Previously this was iterating through the list at least twice (possibly a third as I'm not sure how `*` collections are backed when doing index lookups).